ci/bump_version.sh

```diff
@@ -1,12 +1,32 @@
 #!/bin/bash
 
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[0;33m'
+BLUE='\033[0;34m'
+MAGENTA='\033[0;35m'
+CYAN='\033[0;36m'
+GRAY='\033[0;37m'
+
+# Reset
+NC='\033[0m'
+
 if [ $# != 1 ]
 then
      echo "Please provide version as the first argument"
      exit 1
 fi
 
 version="$1"
+regex="^[0-9]+\.[0-9]+\.[0-9]+$"
```

Comment:
What if somebody supplies numbers with leading zeros or even multiple zeros? For example `000.004.4`. Do we want to have this the new version? Can you find a way how to treat it with an error or how to delete those zeros?

For a single number (so you'd have to repeat the pattern three times) maybe: `0|[1-9][0-9]+?`

Reply:
Well, we can also use this regex

```
^(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)(?:-(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$
```

This is a widely accepted regex for Semantic Versioning followed by the Rust community.

I used this simple regex because it's easy to understand, and this is good enough for the version numbers we use at Masq.

Reply:
Please use this regex as requires more precise syntax.

^(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)$